File: 2nd_quarter/week06/Whaeun/비밀지도.cpp
#include <string>
#include <vector>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    vector<int> arr;
    
    
    for(int i = 0; i < n; i++){
        arr.push_back(arr1[i] | arr2[i]);
    }
    
    string s;
    
    for(int j = 0; j < n; j++){
        for (int i = n-1; i >= 0; i--){
            int temp = (arr[j] >> i) & 1;
            if(temp == 0){
                s += " ";
            }
            else{
                s += "#";
            }
        }

        answer.push_back(s);
        s = "";
    }
    
    return answer;
}

File: 2nd_quarter/week08/Jose/비밀지도.py
def solution(n, arr1, arr2):
    last_answer = []
    answer = [[" " for _ in range(n)] for _ in range(n)]
    for idex, i in enumerate(arr1):
        temp = bin(i)[2:]
        if len(temp) < n:
            for _ in range(n - len(temp)):
                temp = "0" + temp
        
        for jdex, j in enumerate(temp):
            if j == "1": 
                answer[idex][jdex] = "#"
                
    for idex, i in enumerate(arr2):
        temp = bin(i)[2:]
        if len(temp) < n:
            for _ in range(n - len(temp)):
                temp = "0" + temp
                
        for jdex, j in enumerate(temp):
            if j == "1": 
                answer[idex][jdex] = "#"
        
        
    for an in answer:
        last_answer.append("".join(an))
    
    return last_answer
